client/purplecaffeine/core.py

```diff
@@ -160,23 +168,52 @@ def __exit__(self, exc_type, exc_val, exc_tb):
 class BaseBackend:
     """Base backend class."""
 
-    def save(self, name: str, trial: Trial):
+    def save(self, trial: Trial):
         """Saves given trial.
 
         Args:
-            name: name of the trial
             trial: encode trial to save
         """
-        raise NotImplementedError
+        try:
+            requests.post(
+                Configuration.API_FULL_URL,
+                headers=Configuration.API_HEADERS,
+                json=json.dumps(trial.__dict__, cls=TrialEncoder, indent=4),
+                timeout=Configuration.API_TIMEOUT,
+            )
+        except Exception as curl_error:  # pylint: disable=broad-except
+            raise ValueError("Bad request.") from curl_error
+
+        return trial.name
+
+    def get(self, trial_id: int) -> Trial:
+        """Returns trial by name.
+
+        Args:
+            trial_id: trial id
+
+        Returns:
+            trial: object of a trial
+        """
+        try:
+            curl_req = requests.get(
+                f"{Configuration.API_FULL_URL}/{trial_id}/",
+                headers=Configuration.API_HEADERS,
+                timeout=Configuration.API_TIMEOUT,
+            )
+        except Exception as curl_error:  # pylint: disable=broad-except
+            raise ValueError("Bad request.") from curl_error
+
+        return Trial(**json.load(curl_req.json(), cls=TrialDecoder))
 
     def list(
         self,
-        query: Optional[str] = None,
+        query: Optional[str] = None,  # pylint: disable=unused-argument
         limit: Optional[int] = None,
         offset: Optional[int] = None,
-        **kwargs,
-    ) -> List["Trial"]:
-        """Returns list of trails.
+        **kwargs,  # pylint: disable=unused-argument
```

Comment:
@IceKhan13 how did you do here :
https://github.com/IceKhan13/purplecaffeine/blob/main/client/purplecaffeine/core.py#L243

To not have to put the pylint disable ?

Reply:
kwargs should be skipped by pylint